wgpu-hal/src/dx12/adapter.rs

```diff
@@ -377,6 +389,8 @@ impl super::Adapter {
                         .min(crate::MAX_VERTEX_BUFFERS as u32),
                     max_vertex_attributes: d3d12_ty::D3D12_IA_VERTEX_INPUT_RESOURCE_SLOT_COUNT,
                     max_vertex_buffer_array_stride: d3d12_ty::D3D12_SO_BUFFER_MAX_STRIDE_IN_BYTES,
+                    min_subgroup_size: features1.WaveLaneCountMin,
```

Comment:
I've observed a reported value of 16 on Intel GPUs that can actually produce a subgroup size of 8, so best to hardcode this at 4.

The proposal's guidance is, I believe, incorrect. There may be some situations in which the reported values may be considered reliable (possibly in practice newer drivers, conjecturing those that support SM6.6, but I think empirical testing would be needed).